src/Contact_Supplier.php

```diff
@@ -125,108 +119,85 @@ public static function showForContact(Contact $contact)
-            ];
-            Html::showMassiveActions($massiveactionparams);
-        }
-        echo "<table class='tab_cadre_fixehov'>";
-        $header_begin  = "<tr>";
-        $header_top    = '';
-        $header_bottom = '';
-        $header_end    = '';
-        if ($canedit && $number) {
-            $header_top    .= "<th width='10'>" . Html::getCheckAllAsCheckbox('mass' . __CLASS__ . $rand);
-            $header_top    .= "</th>";
-            $header_bottom .= "<th width='10'>" . Html::getCheckAllAsCheckbox('mass' . __CLASS__ . $rand);
-            $header_bottom .= "</th>";
-        }
-        $header_end .= "<th>" . htmlescape(Supplier::getTypeName(1)) . "</th>";
-        $header_end .= "<th>" . htmlescape(Entity::getTypeName(1)) . "</th>";
-        $header_end .= "<th>" . htmlescape(SupplierType::getTypeName(1)) . "</th>";
-        $header_end .= "<th>" . htmlescape(Phone::getTypeName(1)) . "</th>";
-        $header_end .= "<th>" . __s('Fax') . "</th>";
-        $header_end .= "<th>" . __s('Website') . "</th>";
-        $header_end .= "</tr>";
-        echo $header_begin . $header_top . $header_end;
-
-        if ($number > 0) {
-            Session::initNavigateListItems(
-                'Supplier',
-                //TRANS : %1$s is the itemtype name,
-                              //        %2$s is the name of the item (used for headings of a list)
-                                        sprintf(
-                                            __('%1$s = %2$s'),
-                                            Contact::getTypeName(1),
-                                            $contact->getName()
-                                        )
-            );
-
-            foreach ($suppliers as $data) {
-                $assocID = $data["linkid"];
-                Session::addToNavigateListItems('Supplier', $data["id"]);
-                $website           = $data["website"];
-
-                if (!empty($website)) {
-                    $website = $data["website"];
-
-                    if (!preg_match("?https*://?", $website)) {
-                        $website = "http://" . $website;
-                    }
-                    $website = "<a target=_blank href='$website'>" . $data["website"] . "</a>";
+        $entries = [];
+        $suppliertype_cache = [];
+        $entity_cache = [];
+        foreach ($suppliers as $data) {
+            $website           = $data["website"];
+
+            if (!empty($website)) {
+                $website = $data["website"];
+
+                if (!preg_match("?https*://?", $website)) {
+                    $website = "http://" . $website;
                 }
+                $website = "<a target=_blank href='$website'>" . htmlescape($data["website"]) . "</a>";
+            }
+            $supplier_link = "<a href='" . Supplier::getFormURLWithID($data["id"]) . "'>" .
+               htmlescape(Dropdown::getDropdownName("glpi_suppliers", $data["id"])) . "</a>";
```

Comment:
This will permit to render the link only if the current user has access to the supplier.
```suggestion
            $supplier_link = $supplier->getLink();
```